Tests/SwiftLintFrameworkTests/ImplicitlyUnwrappedOptionalRuleTests.swift

```diff
@@ -23,4 +23,28 @@ class ImplicitlyUnwrappedOptionalRuleTests: SwiftLintTestCase {
         verifyRule(description, ruleConfiguration: ["mode": "all"],
                    commentDoesntViolate: true, stringDoesntViolate: true)
     }
+
+    func testImplicitlyUnwrappedOptionalRuleWarnsOnOutletsInWeakMode() {
+        let baseDescription = ImplicitlyUnwrappedOptionalRule.description
+        let triggeringExamples = [
+            Example("private weak var label: UILabel!"),
+            Example("weak var label: UILabel!"),
+            Example("@objc weak var label: UILabel!")
+        ]
+
+        let nonTriggeringExamples = [
+            Example("if !boolean {}"),
+            Example("@IBOutlet private var label: UILabel!"),
+            Example("@IBOutlet var label: UILabel!"),
+            Example("@IBOutlet weak var label: UILabel!"),
+            Example("var label: UILabel!"),
+            Example("let int: Int!")
+        ]
+
+        let description = baseDescription.with(nonTriggeringExamples: nonTriggeringExamples)
+            .with(triggeringExamples: triggeringExamples)
+
+        verifyRule(description, ruleConfiguration: ["mode": "weak_except_iboutlets"],
+                   commentDoesntViolate: true, stringDoesntViolate: true)
```

Comment:
These arguments are defaults, thus they can be skipped:
```suggestion
        verifyRule(description, ruleConfiguration: ["mode": "weak_except_iboutlets"])
```